shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/LoginUseCaseImpl.kt

```diff
@@ -1,41 +1,54 @@
 package io.newm.shared.internal.implementations
 
-import io.newm.shared.login.repository.LogInRepository
-import io.newm.shared.login.repository.OAuthData
+import io.newm.shared.internal.repositories.LogInRepository
+import io.newm.shared.internal.repositories.models.OAuthData
+import io.newm.shared.internal.implementations.utilities.mapErrorsSuspend
 import io.newm.shared.public.models.error.KMMException
+import io.newm.shared.public.usecases.ConnectWalletUseCase
 import io.newm.shared.public.usecases.LoginUseCase
-import shared.Notification
-import shared.postNotification
 import kotlin.coroutines.cancellation.CancellationException
 
 internal class LoginUseCaseImpl(
-    private val repository: LogInRepository) : LoginUseCase {
+    private val repository: LogInRepository,
+    private val connectWalletUseCase: ConnectWalletUseCase
+) : LoginUseCase {
     @Throws(KMMException::class, CancellationException::class)
     override suspend fun logIn(email: String, password: String) {
-        return repository.logIn(email = email, password = password)
+        return mapErrorsSuspend {
+            return@mapErrorsSuspend repository.logIn(email = email, password = password)
+        }
     }
 
     @Throws(KMMException::class, CancellationException::class)
     override suspend fun logInWithGoogle(idToken: String) {
-        return repository.oAuthLogin(OAuthData.Google(idToken))
+        return mapErrorsSuspend {
+            return@mapErrorsSuspend repository.oAuthLogin(OAuthData.Google(idToken))
+        }
     }
 
     @Throws(KMMException::class, CancellationException::class)
     override suspend fun logInWithFacebook(accessToken: String) {
-        return repository.oAuthLogin(OAuthData.Facebook(accessToken))
+        return mapErrorsSuspend {
+            return@mapErrorsSuspend repository.oAuthLogin(OAuthData.Facebook(accessToken))
+        }
     }
 
     @Throws(KMMException::class, CancellationException::class)
     override suspend fun logInWithLinkedIn(accessToken: String) {
-        return repository.oAuthLogin(OAuthData.LinkedIn(accessToken))
+        return mapErrorsSuspend {
+            return@mapErrorsSuspend repository.oAuthLogin(OAuthData.LinkedIn(accessToken))
+        }
     }
 
     @Throws(KMMException::class, CancellationException::class)
     override suspend fun logInWithApple(idToken: String) {
-        return repository.oAuthLogin(OAuthData.Apple(idToken))
+        return mapErrorsSuspend {
+            return@mapErrorsSuspend repository.oAuthLogin(OAuthData.Apple(idToken))
+        }
     }
 
     override fun logout() {
+        connectWalletUseCase.disconnect()
         repository.logout()
```

Comment:
The repository.logout() takes care of disconnecting the wallet
```
 fun logout() {
        tokenManager.clearToken()
        connectWalletManager.disconnect()
    }
```

Reply:
I thought we had agreed on opting against the practice of calling use cases within use cases. This decision was made with the desire to uphold a clear separation of concerns, so that each use case is distinctly responsible for its unique business function. This not only enhances the readability and maintainability of our code, but also greatly simplifies the testing process. By enabling each use case to be tested independently, without the intricacies of intertwined dependencies, we're promoting a more modular and robust system.

Reply:
discussed on a call, leaving it here so can call ConnectWalletUseCase and not miss any side effects.